src/main/java/jewellery/inventory/mapper/ProductMapper.java

```diff
@@ -23,23 +26,34 @@ public ProductResponseDto mapToProductResponseDto(Product product) {
 
     ProductResponseDto productResponseDto = new ProductResponseDto();
     productResponseDto.setId(product.getId());
-    if (product.getPartOfSale() != null) {
+    if (product.getPartOfSale() != null && !product.getPartOfSale().getProducts().isEmpty()) {
```

Comment:
It seems to me we shouldn't need to do the second check `!product.getPartOfSale().getProducts().isEmpty()`. If the product is part of a sale, then surely the products of the sale shouldn't be an empty array.